compiler/rustc_mir/src/interpret/intrinsics.rs

```diff
@@ -323,7 +323,11 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 let result = Scalar::from_uint(truncated_bits, layout.size);
                 self.write_scalar(result, dest)?;
             }
-            sym::copy | sym::copy_nonoverlapping => {
+            sym::copy_nonoverlapping => span_bug!(
+                self.cur_span(),
+                "Unexpected copy_nonoverlapping, should have been lowered in lower_intrinsics",
+            ),
```

Comment:
We already have a catchall at the bottom of this match, you can just let it fall there

Reply:
I think that having this matched explicitly because it's a unique case isn't a bad idea but either way works